File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res=[1]*len(nums)
        product=1
        for i in range(1, len(nums)):
            product=product*nums[i-1]
            res[i]=product
            
        product=1
        for i in range(len(nums)-2, -1, -1):
            product=product*nums[i+1]
            res[i]=res[i]*product
            
        return res

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res(nums.size(), 1);
        int prod=1;
        for(int i=1; i<nums.size(); i++){
            prod=prod*nums[i-1];
            res[i]=prod;
        }
        prod=1;
        for(int i=nums.size()-2; i>=0; i--){
            prod=prod*nums[i+1];
            res[i]=res[i]*prod;
        }
        return res;
    }
};